### PR TITLE
SNOW-1572311:add stress test

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-1572311

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

4. (Optional) PR for stored-proc connector:

### CODE REVIEW

File: test/stress/aio/dev_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+psutil
+../..
+matplotlib
```
Comment (sfc-gh-aling) [inline test/stress/aio/dev_requirements.txt:3]: I think you need aiohttp

File: test/stress/aio/e2e_iterator.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        f"PUT file://../stress_test_data/single_chunk_file_* {stage_location} OVERWRITE = TRUE"
+    )
+    print(f"test file uploaded to {stage_location}", await res.fetchall())
+
+
+async def task_fetch_one_row(cursor, table_name, row_count_limit=50000):
+    ret = await (
+        await cursor.execute(f"select * from {table_name} limit {row_count_limit}")
+    ).fetchone()
+    print(ret)
```
Comment (sfc-gh-aling) [inline test/stress/aio/e2e_iterator.py:158]: fetchone just returns on row, I think you need to do a for loop here

File: test/stress/aio/e2e_iterator.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        if is_multiple
+        else f"GET {stage_file} file://../stress_test_data/"
+    )
+    res = await cursor.execute(sql)
+    print(await res.fetchall())
+
+
+def execute_task(task, cursor, table_name, iteration_cnt):
+    for _ in range(iteration_cnt):
+        task(cursor, table_name)
```
Comment (sfc-gh-aling) [inline test/stress/aio/e2e_iterator.py:205]: is this function used?

File: test/stress/aio/util.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        process.memory_info().rss / 1024 / 1024
+    )  # rss is of unit bytes, we get unit in MB
+    period = time.time() - start
+    if count % SAMPLE_RATE == 0:
+        perf_file.write(str(period) + "\n")
+        print(f"execution time {count}")
+        print(f"memory usage: {memory_usage} MB")
+        print(f"execution time: {period} s")
+        memory_file.write(str(memory_usage) + "\n")
+    count += 1
```
Comment (sfc-gh-aling) [inline test/stress/aio/util.py:31]: do you need this?
I think the way you use it doesn't require sampling -- you basically record the start and end time of the whole execution right?

Comment (sfc-gh-yuwang) [reply test/stress/aio/util.py:31]: yes, i want to keep it here to align with sync connector so if we want to sampling and test in the future

**Review by sfc-gh-aling — COMMENTED** [review]
changes LGTM

**Pull request merged**